python/pyspark/pandas/data_type_ops/boolean_ops.py

```diff
@@ -26,3 +46,183 @@ class BooleanOps(DataTypeOps):
     @property
     def pretty_name(self) -> str:
         return 'booleans'
+
+    def add(self, left, right) -> Union["Series", "Index"]:
+        if isinstance(right, numbers.Number):
+            left = left.spark.transform(lambda scol: scol.cast(as_spark_type(type(right))))
+            return column_op(Column.__add__)(left, right)
```

Comment:
I guess `return left + right` should be fine. The `DataTypeOps` related to `left` after converted should know how to handle it better.

Reply:
Makes sense, addition and other forward operations are applied this approach.